core/src/main/java/org/apache/carbondata/core/scan/result/RowBatch.java

```diff
@@ -100,4 +100,24 @@ public int getSize() {
     counter++;
     return row;
   }
+
+  /**
+   * read next batch
+   *
+   * @param batch batch size
+   * @return rows
+   */
+  public List<Object[]> nextBatch(int batch) {
+    if (!hasNext()) {
+      throw new NoSuchElementException();
+    }
+    List<Object[]> row;
+    if (counter + batch > rows.size()) {
+      row = rows.subList(counter, rows.size());
+    } else {
+      row = rows.subList(counter, counter + batch);
+    }
+    counter = counter + batch;
```

Comment:
What if code enters if check (line 116), incrementing a counter by batch size wrong. because we fetched data lesser that batch size (only till row size)

Reply:
ok, optimized